From f5457b5cef1c82e95621b28e36b5e5aaf2a7b603 Mon Sep 17 00:00:00 2001
Message-Id: <f5457b5cef1c82e95621b28e36b5e5aaf2a7b603.1421835210.git.chang-joon.lee@intel.com>
In-Reply-To: <aa9849875fd9575e7de3d51e779ec326100b81c0.1421835210.git.chang-joon.lee@intel.com>
References: <aa9849875fd9575e7de3d51e779ec326100b81c0.1421835210.git.chang-joon.lee@intel.com>
From: John Harrison <John.C.Harrison@Intel.com>
Date: Thu, 27 Nov 2014 15:44:42 +0000
Subject: [PATCH 53/63] FOR_UPSTREAM [VPG]: drm/i915: Add missing trace point
 to LRC execbuff code path

There is a trace point in the legacy execbuffer execution path that is missing
from the execlist path. Trace points are extremely useful for debugging and are
used by various automated validation tests. Hence, this patch adds the missing
trace point back in.

Change-Id: I8bc80a892e5f7effbe3a0883a58a828d59c10c07
For: VIZ-1587
Signed-off-by: John Harrison <John.C.Harrison@Intel.com>
---
 drivers/gpu/drm/i915/intel_lrc.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_lrc.c b/drivers/gpu/drm/i915/intel_lrc.c
index b4f06ad..a6aeb70 100644
--- a/drivers/gpu/drm/i915/intel_lrc.c
+++ b/drivers/gpu/drm/i915/intel_lrc.c
@@ -1502,6 +1502,8 @@ int intel_execlists_submission(struct drm_device *dev, struct drm_file *file,
 			return ret;
 	}
 
+	trace_i915_gem_ring_dispatch(intel_ring_get_request(ring), dispatch_flags);
+
 	i915_gem_execbuffer_move_to_active(vmas, ring);
 	i915_gem_execbuffer_retire_commands(dev, file, ring, batch_obj);
 
-- 
1.7.9.5

